### DISCUSSION TITLE
goreleaser action: idempotency

### DISCUSSION THREAD

**Discussion by till** [goreleaser/goreleaser#3574] [Q&A] [answered]
For context, we usually have a workflow setup with two jobs, one is to build assets (release files, docker images) and one is to deploy. I had to delete a deployment manually today and was looking for a way to re-run my workflow but it fails because goreleaser cannot publish the artefacts **again**.

Can I do something to run the goreleaser action in a way where it would re-run but not error if the assets already exist? E.g. `422` would be discarded and it would carry on?

```
scm releases: failed to publish artifacts: failed to upload consul-registrator_0.1.6_Linux_x86_64.tar.gz after 2 tries: POST https://uploads.github.com/repos/foo/consul-registrator/releases/83536844/assets?name=consul-registrator_0.1.6_Linux_x86_64.tar.gz: 422 Validation Failed [{Resource:ReleaseAsset Field:name Code:already_exists Message:}]
```

**Comment by caarlos0** [comment]
not really, and its on purpose to avoid weird bugs (like checksum missmatches et al)

what you could do is delete the entire release before running running goreleaser if it exists, can probably be done using some github action.

**Comment by till** [comment, marked as answer]
For recreate I used the api to delete everything.

For re-deploy, ended up skipping this step for re-runs based on workflow event. As in if it's not a tag push but e.g. manual dispatch.
